platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -41,40 +40,25 @@ public abstract class ProjectReportsPlugin implements Plugin<Project> {
     @Override
     public void apply(final Project project) {
         project.getPluginManager().apply(ReportingBasePlugin.class);
-        @SuppressWarnings("deprecation")
-        final ProjectReportsPluginConvention convention = project.getObjects().newInstance(org.gradle.api.plugins.internal.DefaultProjectReportsPluginConvention.class, project);
-
-        DeprecationLogger.whileDisabled(new Runnable() {
-            @Override
-            @SuppressWarnings("deprecation")
-            public void run() {
-                project.getConvention().getPlugins().put("projectReports", convention);
-            }
-        });
-
         project.getTasks().register(TASK_REPORT, TaskReportTask.class, taskReportTask -> {
-            taskReportTask.getProjectReportDirectory().convention(project.getLayout().dir(project.provider(() -> DeprecationLogger.whileDisabled(() -> convention.getProjectReportDir()))));
```

Comment:
I don't think we should remove this property convention for getProjectReportadirectory.

The value of this convention is derived from the deprecated Convention object, but getProjectReportDir delegates to the extension.

We should now derive the property convention from the extension